File: my-solutions/0001-two-sum/solution.java
import java.util.Arrays;
import java.util.Map;

public class Solution {
        public int[] twoSum(int[] nums, int target) {
            if (nums.length == 0 || nums.length < 2)
            return new int[]{0, 0};
            
            Map<Integer,Integer> numberPosition = new HashMap<>();
            
            for(int count=0; count < nums.length; count++){
                int difference = target - nums[count];
                
                if(numberPosition.containsKey(nums[count])){
                    
                    return new int[]{numberPosition.get(nums[count]),count};
                }else{
                    numberPosition.put(difference,count);
                }
            }

        return new int[]{0, 0};
        }
    }

File: my-solutions/0001-two-sum/solution.scala
import scala.collection.mutable._

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        
        val valueIndicesMap = new HashMap[Int, Int]()
    val returnValues = new ArrayBuffer[Int]()


    // Iterate the numbers and populate the map
    for ((num, index) <- nums.zipWithIndex) {

      val difference: Int = target - num

      if (valueIndicesMap.isDefinedAt(difference)) {

        returnValues += valueIndicesMap.get(difference).get
        returnValues += index

      }

      valueIndicesMap += (num -> index)
    }

    return returnValues.toArray
        
        
    }
}
